phongo_compat.h

```diff
@@ -206,6 +182,35 @@
 # define PHP_STREAM_CONTEXT(stream) ((php_stream_context*) (stream)->context)
 #endif
 
+#if SIZEOF_PHONGO_LONG == 8
+# define ADD_INDEX_INT64(zval, index, value) add_index_long(zval, index, value)
+# define ADD_ASSOC_INT64(zval, key, value) add_assoc_long(zval, key, value)
+#elif SIZEOF_PHONGO_LONG == 4
+# define ADD_INDEX_INT64(zval, index, value) \
+    if (value > INT32_MAX || value < INT32_MIN) { \
```

Comment:
I found that these macros were inconsistent with those in `bson.c`, which used `INT_MAX` and `INT_MIN`. The `INT32` constants are more explicit.